From b35286b81e2b6166338008cb9330cf54b0d82250 Mon Sep 17 00:00:00 2001
From: Landry Breuil <landry@xfce.org>
Date: Mon, 10 Dec 2018 09:57:58 +0100
Subject: Fix a crash when opening the properties dialog

introduced in dd5397b3, crash witnessed on OpenBSD and FreeBSD - better setting
the mnemonic widget when the widget has been created.
---
 panel-plugin/systemload.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/panel-plugin/systemload.c b/panel-plugin/systemload.c
index 089b704..ee11297 100644
--- panel-plugin/systemload.c.orig	2018-11-29 16:44:33 UTC
+++ panel-plugin/systemload.c
@@ -806,7 +806,6 @@ static void new_monitor_setting(t_global_monitor *glob
     gtk_widget_set_halign (label, GTK_ALIGN_START);
     gtk_widget_set_valign (label, GTK_ALIGN_CENTER);
     gtk_widget_set_margin_start (label, 12);
-    gtk_label_set_mnemonic_widget (GTK_LABEL (label), button);
     gtk_grid_attach (GTK_GRID(subgrid), label, 0, 0, 1, 1);
 
     /* Entry for the optional monitor label */
@@ -822,6 +821,7 @@ static void new_monitor_setting(t_global_monitor *glob
 
     /* Colorbutton to set the progressbar color */
     button = gtk_color_button_new_with_rgba(colorvar);
+    gtk_label_set_mnemonic_widget (GTK_LABEL (label), button);
     gtk_widget_set_halign(button, GTK_ALIGN_START);
     g_object_set_data(G_OBJECT(button), "colorvar", colorvar);
     g_signal_connect(G_OBJECT(button), "color-set",
